google/cloud/firestore_v1/async_query.py

```diff
@@ -412,3 +460,18 @@ async def get_partitions(
             start_at = cursor
 
         yield QueryPartition(self, start_at, None)
+
+
+class AsyncStreamIterator(abc.AsyncIterator):
+
+    def __init__(self, response_generator):
+        self._generator = response_generator
+
+    def __aiter__(self):
+        return self._generator
+    
+    def __anext__(self):
```

Comment:
I think you shouldn't need to implement the next functions, only iter/aiter. But I could be wrong

If we do need __anext__ as well, you can probably just do
```
def __anext__(self):
  return self._generator.__anext__()
```

Reply:
Thanks, this looks much more succinct! I think we will need to keep the next methods, because it's one of the methods that generator has.